context.go

```diff
@@ -214,27 +221,42 @@ func (c *Context) generateAuthTag(buf, sessionAuthTag []byte) ([]byte, error) {
 	// - Authenticated portion of the packet is everything BEFORE MKI
 	// - k_a is the session message authentication key
 	// - n_tag is the bit-length of the output authentication tag
-	// - ROC is already added by caller (to allow RTP + RTCP support)
-	mac := hmac.New(sha1.New, sessionAuthTag)
+	c.srtpSessionAuth.Reset()
 
-	if _, err := mac.Write(buf); err != nil {
+	if _, err := c.srtpSessionAuth.Write(buf); err != nil {
 		return nil, err
 	}
 
-	return mac.Sum(nil)[0:10], nil
+	// For SRTP only, we need to hash the rollover counter.
+	rocRaw := []byte{byte(roc >> 24), byte(roc >> 16), byte(roc >> 8), byte(roc)}
```

Comment:
How much of a performance loss/gain do we have from this bit shifting? I would prefer things be explicit, as much as the performance matters we will eventually get crushed when people don't want to touch/contribute I think.

Reply:
It just saves needing to clear the memory to zero, so basically no performance difference. I was trying to be clever and save a line of code.